godot-codegen/src/util.rs

```diff
@@ -253,98 +281,86 @@ pub fn make_enum_definition(enum_: &Enum) -> TokenStream {
+                //         )*
+                //         _ => None,
+                //     }
+                // }
+
+                fn try_from_ord(ord: i32) -> Option<Self> {
+                    match ord {
+                        #( ord @ #unique_ords )|* => Some(Self { ord }),
+                        _ => None,
+                    }
                 }
-            }
-        }
-    };
 
-    let mut derives = vec!["Copy", "Clone", "Eq", "PartialEq", "Debug", "Hash"];
-
-    if enum_.is_bitfield {
-        derives.push("Default");
-    }
-
-    let index_enum_impl = if let Some(enum_max) = try_count_index_enum(enum_) {
-        quote! {
-            impl crate::obj::IndexEnum for #enum_name {
-                const ENUMERATOR_COUNT: usize = #enum_max;
+                fn ord(self) -> i32 {
+                    self.ord
+                }
             }
-        }
-    } else {
-        TokenStream::new()
+        };
     };
 
-    let derives = derives.into_iter().map(ident);
-
     // Enumerator ordinal stored as i32, since that's enough to hold all current values and the default repr in C++.
     // Public interface is i64 though, for consistency (and possibly forward compatibility?).
-    // TODO maybe generalize GodotFfi over EngineEnum trait
+    // Bitfield ordinals are stored as u64. See also: https://github.com/godotengine/godot-cpp/pull/1320
     quote! {
         #[repr(transparent)]
         #[derive(#( #derives ),*)]
         pub struct #enum_name {
-            ord: i32
+            ord: #enum_ord_type
         }
         impl #enum_name {
             #(
                 #enumerators
             )*
         }
-        impl crate::obj::EngineEnum for #enum_name {
-            // fn try_from_ord(ord: i32) -> Option<Self> {
-            //     match ord {
-            //         #(
-            //             #matches
-            //         )*
-            //         _ => None,
-            //     }
-            // }
 
-            #try_from_ord
+        #engine_impl
+
+        #bitfield_ops
 
-            fn ord(self) -> i32 {
-                self.ord
-            }
-        }
         #index_enum_impl
 
         impl crate::builtin::meta::GodotConvert for #enum_name {
-            type Via = i32;
+            type Via = #enum_ord_type;
         }
 
         impl crate::builtin::meta::ToGodot for #enum_name {
             fn to_godot(&self) -> Self::Via {
-                <Self as crate::obj::EngineEnum>::ord(*self)
+                #enum_or_bitfield_self::ord(*self)
             }
         }
 
         impl crate::builtin::meta::FromGodot for #enum_name {
             fn try_from_godot(via: Self::Via) -> std::result::Result<Self, crate::builtin::meta::ConvertError> {
-                <Self as crate::obj::EngineEnum>::try_from_ord(via)
+                #enum_or_bitfield_self::try_from_ord(via)
                     .ok_or_else(|| crate::builtin::meta::FromGodotError::InvalidEnum.into_error(via))
```

Comment:
Could you add some docs to [`FromGodotError::InvalidEnum`](https://github.com/godot-rust/gdext/blob/master/godot-core/src/builtin/meta/godot_convert/convert_error.rs#L148-L148) mentioning very briefly that this also includes bitfields?

Reply:
Oops, yeah, makes sense.

Although I'm not quite sure on how we should phrase this, other than: `/// InvalidEnum is also used by bitfields.` 😅